docs/development.md

```diff
@@ -0,0 +1,46 @@
+# Development Environment
+
+This document describes the agreed-upon environment for development. For ease of maintenance, it does not aim to be a
+click by click description, and it assumes familiarity with Windows, Linux, and search engines.
+
+The standard environment is:
+1. Windows
+2. Windows Subsystem for Linux 2 (WSL)
+3. Ubuntu latest in WSL
+4. Visual Studio Code (VS Code)
+
+Other configurations are not prohibited, but keep in mind that when problems arise there will be less support from the
+local community.
+
+## Installation
+
+### 1. Windows
+
+Should be installed for you - easy!
+
+### 2. Windows Subsystem for Linux 2 (WSL)
+
+Follow Microsoft's instructions.
+
+### 3. Ubuntu latest in WSL
+
+- Install Ubuntu latest.
+- `sudo apt update`
+- `sudo apt upgrade`
+- Ubuntu 19.10 comes with Python 3.6, so `sudo apt install python3.7 python3.7-dev`
+- Run `python3 -V` and if it's still 3.6.x then update the link to go to 3.7.
+- `sudo apt install python3-pip`
+- `sudo apt install python3.7-venv`
+- Download the oc client tools matching the current version of OpenShift, and put the *oc* binary in your *~/bin*
+directory.
+
+### 4. Visual Studio Code (VS Code)
+
+- Install Visual Studio code.
+- Install the *Python* extension.
+- Go to File > Preferences > Settings and set the following:
+  - editor.formatOnSave: `true`
```

Comment:
Tried to turn on formatOnSave with other settings as default, for vue file, it removes space before javascript function parentheses, which will cause compile error.  There will be a batch of settings for javascript and typescript format to avoid this happen. For example: 
"javascript.format.insertSpaceBeforeFunctionParenthesis": true
"typescript.format.insertSpaceBeforeFunctionParenthesis": true
etc. 

Maybe having a settings.json with all configuration for Python, Vue.js, etc. will be helpful.

Reply:
Oh, hmm. We're not using JavaScript, so I never thought of that. O'll look into it, thank you!

Reply:
typescript has similar issue.

Reply:
Oh, hmm. Thanks, I'll look into it.

Reply:
And css too. You can try to open a .vue file from a project and save it, you'll see what has been changed by auto format. quit a bit format change with other settings as default.

Reply:
OK, I'll look into all this. We do need a common formatter that we're all using, so there will be some work to format everything and commit it. We can't have compile errors, though. That I'll need to sort out.